### DISCUSSION TITLE
Interest in upstreaming several rules

### DISCUSSION THREAD

**Discussion by GhostLyrics** [ansible/ansible-lint#1754] [New rules]
Hey. I've written several rules for internal use at work and was wondering if there is any interest in having them in the official distribution. 

## Descriptions

These follow the format name - reasoning.

- **copy-instead-of-apt-key** - apt_key is deprecated in Debian because by default it always puts the key in the globally trusted keyring which can be considered insecure
- **mysql-instead-of-command** - there is a well-maintained community module `community.mysql` for this which is likely to produce better results in most cases
- **mysql-instead-of-shell** - (same as above)
- **template-instead-of-apt-repository** - apt_repository doesn't do proper string-matching or implicit templates; instead when changing strings and re-applying, you will have created a duplicate entry
- **template-instead-of-copy** - this rule prevents using `copy` with `content` which might be highly opinionated, but was triggered by the following line in the Ansible docs:
> If you need variable interpolation in copied files, use the ansible.builtin.template module. Using a variable in the content field will result in unpredictable output.

Please let me know what you think. I'm absolutely open to suggestions in regards to the `severity` and `tags` sections of the rules, I just filled them with what seemed reasonable for our project for now.

**Comment by Zocker1999NET** [comment]
I would highly recommend to add template-instead-of-copy to upstream. In older docs, it was even recommended to use copy content when one wanted to inline the template. I read the new warning just by chance. See ansible/ansible#50580 for more reference why the recommendation changed. I wanted to propose it myself after reading the warning.

**Reply by GhostLyrics** [reply]
[MR sent](https://github.com/ansible/ansible-lint/pull/2346).
